pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -198,6 +201,13 @@ public void init(SegmentGeneratorConfig config, SegmentCreationDataSource dataSo
     // Create a temporary directory used in segment creation
     _tempIndexDir = new File(indexDir, "tmp-" + UUID.randomUUID());
     LOGGER.debug("tempIndexDir:{}", _tempIndexDir);
+
+    _upsertSnapshotEnabled = _config.isUpsertSnapshotEnabled();
```

Comment:
We don't have the valid doc id info during segment creation. The snapshot has to be taken periodically during consumption. In the current implementation, no valid doc is added

Reply:
Thanks @Jackie-Jiang i planned to create another PR for TTL-related config and logics. Do you think I should add that in the PR?

Reply:
Should we do this in `build` method? I think this shall be enabled with TTL, as it applies only to the segments outside TTL, otherwise the valid doc id will keep changing